File: src-ts/Commands/Music/play.ts
import { ApplicationCommandOptionType } from "discord.js";
import { Command } from "../../Interfaces";

export const command: Command = {
  name: "play",
  description: "Adds song to queue",
  category: "Music",
  options: [{
    name: "song",
    description: "Song name or Song URL",
    type: ApplicationCommandOptionType.String,
    required: true
  }],
  aliases: ["p"],
  run: (variables) => {
    variables.client.servers.get(variables.guild_id)?.player.play(variables);
  }
}

File: src/bot/commands/music/play.rs
use crate::{
    bot::commands::{music::handle_vc_connection, Context, Error},
    player::Song,
};

/// Adds song to queue
#[poise::command(
    slash_command,
    prefix_command,
    aliases("p"),
    category = "Music",
    guild_only
)]
pub async fn play(
    ctx: Context<'_>,
    #[description = "Song name or Song URL"]
    #[rest]
    song: String,
) -> Result<(), Error> {
    let server = get_server!(ctx);

    handle_vc_connection(&ctx, &server).await?;

    ctx.defer().await?;

    let mut songs = Song::new(&ctx, song).await?;
    match songs.len() {
        0 => {
            message!(error, ctx, ("An error happened please try again later"); false);
            return Ok(());
        },
        1 =>
            message!(success, ctx, ("**{}** has been added to the queue.", songs[0].title()); false),
        _ => message!(success, ctx, ("**{}** songs added to the queue.", songs.len()); false),
    }
    server.player.play(&mut songs).await;

    Ok(())
}
